zerver/views/streams.py

```diff
@@ -518,27 +532,86 @@ def remove_subscriptions_backend(
     return json_success(request, data=result)
 
 
-def you_were_just_subscribed_message(
-    acting_user: UserProfile, recipient_user: UserProfile, stream_names: Set[str]
+def make_message_for_subscription_change(
+    acting_user: UserProfile,
+    recipient_user: UserProfile,
+    stream_names: Set[str],
+    *,
+    add_subscription: bool = False,
+    remove_subscription: bool = False,
 ) -> str:
     subscriptions = sorted(stream_names)
-    if len(subscriptions) == 1:
-        with override_language(recipient_user.default_language):
+    message: str = ""
+    with override_language(recipient_user.default_language):
+        if add_subscription and len(subscriptions) == 1:
             return _("{user_full_name} subscribed you to the stream {stream_name}.").format(
                 user_full_name=f"@**{acting_user.full_name}|{acting_user.id}**",
                 stream_name=f"#**{subscriptions[0]}**",
             )
-
-    with override_language(recipient_user.default_language):
-        message = _("{user_full_name} subscribed you to the following streams:").format(
-            user_full_name=f"@**{acting_user.full_name}|{acting_user.id}**",
-        )
+        elif remove_subscription and len(subscriptions) == 1:
+            return _("{user_full_name} unsubscribed you from the stream {stream_name}.").format(
+                user_full_name=f"@**{acting_user.full_name}|{acting_user.id}**",
+                stream_name=f"#**{subscriptions[0]}**",
+            )
+        elif add_subscription:
+            message = _("{user_full_name} subscribed you to the following streams:").format(
+                user_full_name=f"@**{acting_user.full_name}|{acting_user.id}**",
+            )
+        else:
+            message = _("{user_full_name} unsubscribed you from the following streams:").format(
```

Comment:
Need prep commit for this branch to be included in test coverage. The api support this, but i couldn't find how to do this in the web app. The original issue is unclear on whether we need this, but as seen in the diff, this behaviour was accounted for previously.

Reply:
If the API supports this, then handling it is the right decision. -- otherwise no message would be sent in that case.